### PR TITLE
#1476 Refactor ComposeViewController

### LINKED ISSUE #1476: ComposeViewController divide into small files
Currently ComposeViewController is about 2000 lines long and hard to read/maintain.
Should be divided into small parts.

### PR DESCRIPTION
**Pull request by ioanmo226**
This PR refactored ComposeViewController

close #1476 // if this PR closes an issue

----------------------------------

**Tests** _(delete all except exactly one)_:
- Does not need tests (refactor only, docs or internal changes)

--------------------------------

### To be filled by reviewers

I have reviewed that this PR... _(tick whichever items you personally focused on during this review)_:
- [x] addresses the issue it closes (if any)
- [x] code is readable and understandable
- [x] is accompanied with tests, or tests are not needed
- [x] is free of vulnerabilities
- [x] is documented clearly and usefully, or doesn't need documentation

### CODE REVIEW

File: FlowCrypt/Controllers/Compose/Extensions/ComposeViewController+Search.swift [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+            .removeDuplicates()
+            .map { [weak self] query -> String in
+                if query.isEmpty {
+                    self?.updateState(with: .main)
+                }
+                return query
+            }
+            .sink(receiveValue: { [weak self] in
+                guard $0.isNotEmpty else { return }
+                self?.searchEmail(with: $0)
```
Comment (tomholub) [inline FlowCrypt/Controllers/Compose/Extensions/ComposeViewController+Search.swift:23]: Let's put this in `SetupUI` which can be renamed to `Setup`

File: FlowCrypt/Controllers/Compose/Extensions/ComposeViewController+State.swift [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+import UIKit
+
+// MARK: - State Handling
+extension ComposeViewController {
+    internal func updateState(with newState: State) {
+        if case .searchEmails = newState, !self.shouldDisplaySearchResult {
+            return
+        }
+
```
Comment (tomholub) [inline FlowCrypt/Controllers/Compose/Extensions/ComposeViewController+State.swift:17]: This would belong in `RecipientInput`, probably.

**Review by tomholub — DISMISSED** [review]
Very good. See comments

File: FlowCrypt/Controllers/Compose/Extensions/ComposeViewController+State.swift (hunk 2 above)
Comment (ioanmo226) [reply FlowCrypt/Controllers/Compose/Extensions/ComposeViewController+State.swift:17]: But this function updates recipient list, password section, search results.

File: FlowCrypt/Controllers/Compose/Extensions/ComposeViewController+Search.swift (hunk 1 above)
Comment (ioanmo226) [reply FlowCrypt/Controllers/Compose/Extensions/ComposeViewController+Search.swift:23]: Yeah, I agree

File: FlowCrypt/Controllers/Compose/Extensions/ComposeViewController+State.swift (hunk 2 above)
Comment (tomholub) [reply FlowCrypt/Controllers/Compose/Extensions/ComposeViewController+State.swift:17]: I see. We can leave it as is for now. Later, maybe it needs some renaming to make it clearer what it does.

**Review by tomholub — APPROVED** [review]

File: FlowCrypt/Controllers/Compose/Extensions/ComposeViewController+State.swift (hunk 2 above)
Comment (ioanmo226) [reply FlowCrypt/Controllers/Compose/Extensions/ComposeViewController+State.swift:17]: Ok

**Pull request merged**